### PR TITLE
mat64: add element-wise arithmetic to Vector

### PR DESCRIPTION
**Pull request by kortschak**
Also fix doc comment on Dense DivElem.

@btracey

### CODE REVIEW

**Comment by btracey** [comment]
I don't think this is a good idea. As much as reasonable, we should try to only provide "type safe" methods. For example, here we should see 
````
func ( v Vector) AddVec(a, b Vector)
````
As a vector is the result of adding a vector and a vector, not the result of adding a Matrix and a Matrix.

**Comment by kortschak** [comment]
PTAL

**Comment by kortschak** [comment]
Ignore that. Fixing now.

**Comment by kortschak** [comment]
PTAL

File: mat64/vector.go [hunk 1]
```diff
@@ -103,6 +103,74 @@ func (m *Vector) RawVector() blas64.Vector {
...
+
+	v.reuseAs(ar)
+
+	amat, bmat := a.RawVector(), b.RawVector()
+	for i := 0; i < v.n; i++ {
+		v.mat.Data[i*v.mat.Inc] = amat.Data[i*amat.Inc] - bmat.Data[i*bmat.Inc]
+	}
+}
+
+// MulElemVec performs element-wise multiplication of a and b, placing the result
```
Comment (btracey) [inline mat64/vector.go]: Comment missing.

File: mat64/vector.go [hunk 2]
```diff
@@ -103,6 +103,74 @@ func (m *Vector) RawVector() blas64.Vector {
...
+
+	v.reuseAs(ar)
+
+	amat, bmat := a.RawVector(), b.RawVector()
+	for i := 0; i < v.n; i++ {
+		v.mat.Data[i*v.mat.Inc] = amat.Data[i*amat.Inc] * bmat.Data[i*bmat.Inc]
+	}
+}
+
+// DivElemVec performs element-wise division of a by b, placing the result
```
Comment (btracey) [inline mat64/vector.go]: Same here.

**Comment by btracey** [comment]
LGTM with comments fixed.

File: mat64/vector.go (hunk 1 above)
Comment (kortschak) [inline mat64/vector.go]: Fixed.

File: mat64/vector.go (hunk 2 above)
Comment (kortschak) [inline mat64/vector.go]: Fixed.

Thanks for your patience.

**Comment by kortschak** [comment]
travis pr tests failed because it hadn't pulled the pr refs before the merge and github deletes them at merge.

**Pull request merged**